archetypes/helidon/src/main/archetype/mp/custom/files/src/test/java/__pkg__/TestCORS.java.mustache

```diff
@@ -1,141 +1,108 @@
 package {{package}};
 
-import io.helidon.common.http.Headers;
 import io.helidon.common.http.Http;
 import io.helidon.config.Config;
-import io.helidon.microprofile.server.Server;
-import io.helidon.reactive.webclient.WebClient;
-import io.helidon.reactive.webclient.WebClientRequestBuilder;
-import io.helidon.reactive.webclient.WebClientRequestHeaders;
-import io.helidon.reactive.webclient.WebClientResponse;
-import io.helidon.reactive.webclient.WebClientResponseHeaders;
 import io.helidon.cors.CrossOriginConfig;
+import io.helidon.microprofile.tests.junit5.HelidonTest;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.Invocation;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Response;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-import java.util.List;
-import java.util.Optional;
-
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
 
+@HelidonTest
 class TestCORS {
 
-    private static WebClient client;
-    private static Server server;
+    @Inject
+    private WebTarget target;
+    private static final String ALLOW_RESTRICTED_HEADERS = "sun.net.http.allowRestrictedHeaders";
```

Comment:
Where is that coming from ?

Reply:
You can see it from `BaseCrossOriginTest`. The system property is needed to be set to allow restricted header with Jersey client.

Reply:
How did it work without that ?

Reply:
I think the reason is that reactive webClient is using `HelidonConnector` while now it is using the Jersey default `HttpUrlConnector` where this property manage restricted headers.